srcdeps-core-config-yaml/src/main/java/org/srcdeps/config/yaml/writer/YamlWriterVisitor.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2015-2016 Maven Source Dependencies
+ * Plugin contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.srcdeps.config.yaml.writer;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.Writer;
+
+import org.srcdeps.core.config.Configuration;
+import org.srcdeps.core.config.tree.ContainerNode;
+import org.srcdeps.core.config.tree.ListNode;
+import org.srcdeps.core.config.tree.Node;
+import org.srcdeps.core.config.tree.ScalarNode;
+import org.srcdeps.core.config.tree.Visitor;
+import org.srcdeps.core.config.tree.walk.AbstractVisitor;
+
+/**
+ * A {@link Visitor} that serializes the given tree to a {@link Writer}.
+ *
+ * @author <a href="https://github.com/ppalaga">Peter Palaga</a>
+ */
+public class YamlWriterVisitor extends AbstractVisitor implements Closeable {
```

Comment:
We use SnakeYaml to parse, but as I said above, SnakeYAML ignores comments by design. I plan to use this `YamlWriterVisitor` to output an auto-generated `srcdeps.yaml` files in the future and I think I'll need comments there. So while we'll probably keep using SnakeYAML for reading YAML files, we will not use it for writing them.